ydb/services/ydb/ydb_login_ut.cpp

```diff
@@ -0,0 +1,158 @@
+        }
+    }
+
+private:
+    NKikimrConfig::TAppConfig InitAuthSettings(bool isLoginAuthenticationEnabled = true) {
+        NKikimrConfig::TAppConfig appConfig;
+        auto authConfig = appConfig.MutableAuthConfig();
+
+        authConfig->SetUseBlackBox(false);
+        authConfig->SetUseLoginProvider(true);
+        authConfig->SetEnableLoginAuthentication(isLoginAuthenticationEnabled);
+        appConfig.MutableDomainsConfig()->MutableSecurityConfig()->SetEnforceUserTokenRequirement(true);
+        appConfig.MutableFeatureFlags()->SetAllowYdbRequestsWithoutDatabase(false);
+        appConfig.MutableFeatureFlags()->SetCheckDatabaseAccessPermission(true);
+
+        return appConfig;
+    }
+
+    TDriverConfig GetDriverConfig(ui16 grpcPort) {
+        TDriverConfig config;
+        config.SetEndpoint("localhost:" + ToString(grpcPort));
+        return config;
+    }
+
+private:
+    TKikimrWithGrpcAndRootSchemaWithAuth Server;
+    NYdb::TDriver Connection;
+    NConsoleClient::TDummyClient Client;
+};
+
+} // namespace
+
+
+Y_UNIT_TEST_SUITE(TGRpcAuthentication) {
+    const TString User = "user";
+    const TString Password = "UserPassword";
+
+    Y_UNIT_TEST(ValidCredentials) {
+        TLoginClientConnection loginConnection;
+        loginConnection.CreateUser(User, Password);
+        loginConnection.ModifyACL(true, User, NACLib::EAccessRights::ConnectDatabase | NACLib::EAccessRights::DescribeSchema);
+        auto factory = CreateLoginCredentialsProviderFactory({.User = User, .Password = Password});
+        auto loginProvider = factory->CreateProvider(loginConnection.GetCoreFacility());
+        TString token;
+        UNIT_ASSERT_NO_EXCEPTION(token = loginProvider->GetAuthInfo());
+        UNIT_ASSERT(!token.empty());
+
+        loginConnection.Ls(token, "");
+
+        loginConnection.Stop();
+    }
+
+    Y_UNIT_TEST(InvalidPassword) {
+        TLoginClientConnection loginConnection;
+        loginConnection.CreateUser(User, Password);
+
+        auto factory = CreateLoginCredentialsProviderFactory({.User = User, .Password = "WrongPassword"});
+        auto loginProvider = factory->CreateProvider(loginConnection.GetCoreFacility());
+        UNIT_ASSERT_EXCEPTION_CONTAINS(loginProvider->GetAuthInfo(), yexception, "Invalid password");
+
+        loginConnection.Stop();
+    }
+
+    Y_UNIT_TEST(DisableBuiltinAuthMechanism) {
+        TLoginClientConnection loginConnection(false);
+
+        auto factory = CreateLoginCredentialsProviderFactory({.User = User, .Password = Password});
+        auto loginProvider = factory->CreateProvider(loginConnection.GetCoreFacility());
+        TStringBuilder expectedErrorMessage;
+        UNIT_ASSERT_EXCEPTION_CONTAINS(loginProvider->GetAuthInfo(), yexception, "Login authentication is disabled");
+
+        loginConnection.Stop();
+    }
+
+    Y_UNIT_TEST(NoDescribeRights) {
```

Comment:
In this test user has not connect rights also. Are you sure you're checking for the absence of the correct right?

Reply:
fc7e6eaa976a27155ef2f9f2af952b1c0e151a03